Raven.Database/Raft/Storage/Handlers/ReplicationStateCommandHandler.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Rachis;
+using Raven.Abstractions.Extensions;
+using Raven.Bundles.Replication.Data;
+using Raven.Database.Raft.Commands;
+using Raven.Database.Server.Tenancy;
+
+namespace Raven.Database.Raft.Storage.Handlers
+{
+    public class ReplicationStateCommandHandler : CommandHandler<ReplicationStateCommand>
+    {
+        public ReplicationStateCommandHandler(DocumentDatabase database, DatabasesLandlord landlord,RaftEngine engine) : base(database, landlord)
+        {
+            MaxReplicationLatency = database.Configuration.Cluster.MaxReplicationLatency;
+            Engine = engine;
+        }
+
+        private RaftEngine Engine { get; set; }
+
+        //default should be 2 minutes
+        public TimeSpan MaxReplicationLatency { get; set; }
+
+        public override void Handle(ReplicationStateCommand command)
+        {
+            //If we are the leader we won't have documents sent from ourself so it is a noop.
+            if (Engine.State == RaftEngineState.Leader)
+                return;
+            var lastChanges = command.DatabaseToLastModified;
+            foreach (var databaseName in lastChanges.Keys)
+            {
+                Task<DocumentDatabase> databaseTask;
+                if(Landlord.TryGetOrCreateResourceStore(databaseName, out databaseTask) == false)
+                    throw new InvalidOperationException($"ReplicationStateCommand could not load database: {databaseName}");
+                var database = databaseTask.Result;
```

Comment:
You are forcing us to load the database and wait for it, why?
Why aren't we saving this in the cluster state (system db) instead of in each individual db?

We need it for the cluster, not for each db